Python/compile.c

```diff
@@ -4194,9 +4194,20 @@ compiler_nameop(struct compiler *c, location loc,
         optype = OP_DEREF;
         break;
     case LOCAL:
-        if (_PyST_IsFunctionLike(c->u->u_ste) ||
-                (PyDict_GetItem(c->u->u_metadata.u_fasthidden, mangled) == Py_True))
+        if (_PyST_IsFunctionLike(c->u->u_ste)) {
             optype = OP_FAST;
+        }
+        else {
+            PyObject *item;
+            if (PyDict_GetItemRef(c->u->u_metadata.u_fasthidden, mangled,
+                                  &item) < 0) {
+                return ERROR;
```

Comment:
`mangled` is leaked here. As well as in other early returns.

Reply:
Oh right, I copied/pasted code around, I didn't notice that there was already a leak. I added an ``error`` label for these code paths.